test/circuit_test.rb

```diff
@@ -95,4 +95,21 @@ class CircuitTest < Minitest::Spec
     expect(j).must_equal 2
     expect(bla).must_equal []
   end
+
+  it "throws an exception if any unknown signal is caught" do
+    exception = assert_raises Trailblazer::Activity::Circuit::IllegalSignalError do
+      circuit = outer
+
+      # stub out fetching next task to return nothing and raise the exception
+      def circuit.next_for(task, signal); nil; end
+
+      circuit.([ {} ])
+    end
+
+    message = "Unrecognized Signal [\"to a\"] returned from step <#{Start}> \nValid signals are: \n\tto a\n\tto b"
```

Comment:
It would be cool if it said "from `ActivityThrowingtheException`"?!?!

Reply:
Updated it @apotonick 👍
Is it cool if we use `exec_context` for getting actual activity ?

Reply:
That's a great idea, didn't even think of it! :+1: